DomainContent/Hub/RainbowThrower/scripts/rainbowThrowerGunSpawner.js

```diff
@@ -28,6 +27,7 @@
         preload: function(entityID) {
             _this.entityID = entityID;
             _this.position = Entities.getEntityProperties(_this.entityID, 'position').position;
```

Comment:
These should get consolidated into a single `getEntityProperties` call that requests both position and rotation